src/vm/peimage.inl

```diff
@@ -481,8 +481,19 @@ inline BOOL PEImage::HasNativeHeader()
         return GetLoadedLayout()->HasNativeHeader();
     else
     {
-        PEImageLayoutHolder pLayout(GetLayout(PEImageLayout::LAYOUT_ANY,LAYOUT_CREATEIFNEEDED));
```

Comment:
Short of asking the callstack at which the exception is thrown, is it thrown during the call at https://github.com/dotnet/coreclr/pull/11341/files#diff-1708de55e5220e56e33f7ddb0f465f9cR1591? If so, should we have the try/catch at that point and then proceed to bind as IL as your change enables?